### ISSUE TITLE
[CI] Windows Provider Unreachable

### ISSUE THREAD

**Issue by JP-Ellis** [pact-foundation/pact-python#639]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/pact-foundation/pact-python/blob/master/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of `pact-python`.
- [X] I have read the console error message carefully (if applicable).

### Description

The provider tests require a provider server to be started. This is done using the Flask framework and by starting the server through a sub-process.

During initialisation, the PyTest fixture polls the `/_test/ping` endpoint until a success is returned. For some reason, the polling never succeeds, resulting in a failure.

### Steps to reproduce

In current `master` branch

### Expected behavior

Connection succeeds, allowing for the test to proceed and succeed

### Actual behavior

Test stalls during the polling, and eventually fails after too many retries.

### Your environment

`master` branch of this repo.

### Self-service

- [X] I'd be willing to fix this bug myself.

**Comment by Nikhil172913832** [comment]
Hi @JP-Ellis,

After analyzing the Windows CI failures in test_match.py, I found that the issue originates from using subprocess.Popen to start the Flask provider (around line 36).

This method is unreliable on Windows due to differences in process spawning and output buffering, which can cause the test to hang while polling the /_test/ping endpoint.

Proposed fix:
Refactor start_provider() to start the Flask provider in a thread instead of a subprocess, with an explicit host and port. This avoids the subprocess synchronization issues on Windows and aligns with the approach already used in test_provider.py.

Let me know if you'd like me to open a PR for this change.

**Comment by JP-Ellis** [comment]
Yeah, I think that's fine. As you've identified, there are existing tests which already work using daemonic threads.

Note that I would probably prefer that a port be dynamically obtained, as opposed to relying on a hard-coded value, in order to avoid any possible conflicts.
